### ISSUE TITLE
Add information about direct and transitive dependencies for iOS and SQLCipher

### ISSUE THREAD

**Issue by cremfert** [simolus3/drift#1480]
When working with iOS and SQLCipher with drift:ffi, other libraries like cached_network_image will override sqlite with their own version, leading to a conflict when using SQLCipher. As a result, the SQLite file couldn't be opened and might get corrupted.

Details can be found here: https://github.com/simolus3/moor/discussions/1479

@davidmartos96 provided a solution for this problem here:
https://github.com/davidmartos96/sqflite_sqlcipher/blob/master/sqflite/README.md#if-using-sqflite-as-direct-or-transitive-dependency

```
dependency_overrides:
  sqflite:
    git:
      url: https://www.github.com/davidmartos96/sqflite_sqlcipher.git
      path: sqflite
      ref: fmdb_override
```

I suggest to put this in the docs as well because it took a lot of time to figure this out. Besides that, thanks for this great library!

**Comment by simolus3** [comment]
> without the changes to the pubspec.yml or the XCode project, it works on the device but actually without encryption!

But wouldn't there be exceptions due to the key pragma not existing? So while this is certainly annoying, it does not look like a security issue to me.

> With dependency_overrides OR other linker flag options: encryption is working and showing

I think the linker flag alone should work in all cases. Even if you do have a dependency on `sqflite`, it shouldn't affect the native build if you tell the linker to prioritize SQLCipher. The linker flag will also work for non-obvious packages like Google mobile ads which depend on sqlite3 natively without depending on `sqflite`.

I definitely agree that this should be made more obvious in the documentation though.

**Comment by cremfert** [comment]
> But wouldn't there be exceptions due to the key pragma not existing? So while this is certainly annoying, it does not look like a security issue to me.

I also thought this, but unfortunately, there is no error message. This made me believe that everything is working fine and I just found out that it isn't when I tried to open the file in a DB browser. This is what I meant by "security issue" as developers might believe that everything works fine including encryption, when it actually doesn't.

> I definitely agree that this should be made more obvious in the documentation though.

Thank you very much and also for your great library! I really appreciate your work.
